### DISCUSSION TITLE
How to backup volumes from helm hook

### DISCUSSION THREAD

**Discussion by jonaskistler** [vmware-tanzu/velero#6967] [Community support Q&A]
Hello ! 

Is there a way to run a Velero backup from a helm hook ? I would like to create a new backup on a pre-upgrade hook. I am using Velero with GCP and scheduled backups and restores are working fine. 

Thanks !

**Comment by mallancrime** [comment]
Yes, you can trigger a Velero backup using a Helm hook. Helm hooks allow you to execute commands at certain points during the Helm lifecycle, such as before or after a release is installed, upgraded, or deleted.

In your case, you want to create a new backup on a pre-upgrade hook. Here's an example of how you might achieve this:

Update your Helm chart:
Add a pre-upgrade hook to your templates directory. Let's call it pre-upgrade-job.yaml. This file might look like:
apiVersion: batch/v1
kind: Job
metadata:
  name: pre-upgrade-backup
spec:
  template:
    metadata:
      name: pre-upgrade-backup
    spec:
      containers:
      - name: velero
        image: velero/velero:latest
        command: ["velero", "backup", "create", "new-backup-name", "--selector", "app=my-app"]
        env:
        - name: AWS_ACCESS_KEY_ID
          valueFrom:
            secretKeyRef:
              name: cloud-credentials
              key: cloud-credentials-access-key-id
        - name: AWS_SECRET_ACCESS_KEY
          valueFrom:
            secretKeyRef:
              name: cloud-credentials
              key: cloud-credentials-secret-access-key
      restartPolicy: Never
This job runs the Velero backup command with the necessary parameters.

Modify your Helm chart:
Update your main deployment.yaml or equivalent to include the hook:

...
spec:
  replicas: 1
  template:
    metadata:
      labels:
        app: my-app
    spec:
      containers:
        - name: my-app
          image: my-app:latest
  ...
  hooks:
    - apiVersion: batch/v1
      kind: Job
      name: pre-upgrade-backup
      weight: 2

The weight attribute determines the order of execution among hooks. You can adjust it as needed.

Deploy or upgrade your Helm release:
When you deploy or upgrade your Helm release, the pre-upgrade hook will trigger the job, and Velero will create a new backup.

Remember to adapt the example to your specific requirements, such as adjusting the image names, labels, and environment variables according to your Helm chart structure and Velero configuration.
Our platform offers a user-friendly interface, allowing you to effortlessly browse and access a multitude of applications tailored to your preferences. Whether you're seeking productivity tools, entertainment apps, or innovative utilities, [NY APK](https://nyapk.com/) is your go-to source. Join our community of Android enthusiasts and elevate your app experience with the latest and most trusted APKs, all at NY APK!"

**Comment by vindiesel56** [comment]
"Is there a gaming equivalent to running a 'Velero backup from a Helm hook'? I'm interested in initiating a game state capture or save point during a 'pre-upgrade' phase. Similar to how Velero ensures data integrity, in gaming, having a seamless process for capturing and restoring game progress could be a game-changer, especially for complex multiplayer scenarios or game world modifications. Any insights or suggestions for implementing such a mechanism in the [android game](https://carparkingmultiplayermodapk.net/) context would be appreciated!"
